### DISCUSSION TITLE
Can I immediately cancel all TaskInstanceIds not in my list or select all TaskInstanceIds from database?

### DISCUSSION THREAD

**Discussion by hana-krizova** [kagkarlsson/db-scheduler#416] [Q&A] [answered]
Hello Kagkarlsson and community,

please, can I immediately cancel all task instance Ids except that I have in my input list? I would like to have a clean log.

I know not longer supported task instances from customer list, which I use like my program input. So for this purpose it would be enough and may be better when Scheduler provides all task instance Ids from the database (something like this: select task_name, task_instance from $tableName order by execution_time asc)?

Your sincerely
Hana Křížová

**Comment by kagkarlsson** [comment, marked as answer]
TBH I am not sure I understand the question here

You can already fetch scheduled executions using `scheduler.getScheduledExecutions(...)`

**Reply by hana-krizova** [reply]
Yes, thank you very much, SchedulerClient#getScheduledExecutions() is exactly what I need.
I didn't realized that just non picked executions are enough for me - also because picked is not possible to cancel.
